spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/RdbmsTaskDefinitionRepository.java

```diff
@@ -19,313 +19,48 @@
-
-	public static final String LIST_OF_NAMES = "listnames";
-
-	public static final String TASK_DEFINITION_NAME_WHERE_CLAUSE = "where DEFINITION_NAME = ? ";
-
-	public static final String TASK_DEFINITION_NAME_IN_CLAUSE = "where DEFINITION_NAME in ( :" + LIST_OF_NAMES + ") ";
-
-	private static final String FIND_ALL_DEFINITIONS = "SELECT " + SELECT_CLAUSE + "FROM "
-			+ FROM_CLAUSE;
-
-	private static final String FIND_DEFINITIONS_WHERE_IN_CLAUSE = FIND_ALL_DEFINITIONS
-			+ TASK_DEFINITION_NAME_IN_CLAUSE;
-
-	private static final String SAVE_TASK_DEFINITION = "INSERT into %PREFIX%DEFINITION"
-			+ "(DEFINITION_NAME, DEFINITION)"
-			+ "values (?, ?)";
-
-	private static final String TASK_DEFINITION_COUNT = "SELECT COUNT(*) FROM "
-			+ FROM_CLAUSE;
-
-	private static final String TASK_DEFINITION_COUNT_BY_NAME = "SELECT COUNT(*) FROM " + FROM_CLAUSE
-			+ TASK_DEFINITION_NAME_WHERE_CLAUSE;
-
-	private static final String FIND_TASK_DEFINITION_BY_NAME = FIND_ALL_DEFINITIONS + TASK_DEFINITION_NAME_WHERE_CLAUSE;
-
-	private static final String DELETE_FROM_TASK_DEFINITION = "DELETE FROM " + FROM_CLAUSE;
-
-	private static final String DELETE_FROM_TASK_DEFINITION_BY_NAME = DELETE_FROM_TASK_DEFINITION
-			+ TASK_DEFINITION_NAME_WHERE_CLAUSE;
-
-	private String tablePrefix = DEFAULT_TABLE_PREFIX;
-
-	private String findAllDefinitionsQuery;
-
-	private String fromClauseQuerySegment;
-
-	private String saveTaskDefinitionStatement;
-
-	private String taskDefinitionCountQuery;
-
-	private String taskDefinitionCountByNameQuery;
-
-	private String findTaskDefinitionByNameQuery;
-
-	private String deleteFromTaskDefinitionStatement;
-
-	private String deleteFromTaskDefinitionByNameStatement;
-
-	private String findDefinitionsWhereInClause;
-
-	private JdbcOperations jdbcTemplate;
-
-	private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
-
-	private DataSource dataSource;
-
-	private Map<String, Order> orderMap;
+public class RdbmsTaskDefinitionRepository extends AbstractRdbmsKeyValueRepository<TaskDefinition> implements TaskDefinitionRepository {
 
 	public RdbmsTaskDefinitionRepository(DataSource dataSource) {
-		Assert.notNull(dataSource);
-		this.jdbcTemplate = new JdbcTemplate(dataSource);
-		this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
-		this.dataSource = dataSource;
-		this.orderMap = new TreeMap<>();
-		this.orderMap.put("DEFINITION_NAME", Order.ASCENDING);
-		updateQueryPrefixes();
-	}
-
-	@Override
-	public Iterable<TaskDefinition> findAll(Sort sort) {
-		Assert.notNull(sort, "sort must not be null");
-		Iterator<Sort.Order> iter = sort.iterator();
-		String query = findAllDefinitionsQuery  + "ORDER BY " ;
-
-		while (iter.hasNext()) {
-			Sort.Order order = iter.next();
-			query = query + order.getProperty() + " " + order.getDirection();
-			if (iter.hasNext()) {
-				query = query + ", ";
+		super(dataSource, "TASK_", "DEFINITIONS", new RowMapper<TaskDefinition>() {
```

Comment:
I was thinking the reverse as we can specify the `prefix` and `suffix` for the table here in the repository class directly rather than specifying where they are instantiated. But, I am ok with both the ways.